### PR TITLE
Fix get_numba_envvar

### LINKED ISSUE #6601: get_numba_envvar doesn't seem to return the right value
If the env var value is found, Instead of returning the value, the function returns the default value.

https://github.com/numba/numba/blob/42050e4a306e53f332fbaecf45248c54fae752e5/numba/cuda/envvars.py#L32

### PR DESCRIPTION
**Pull request by sklam**
Fixes #6601

### CODE REVIEW

File: numba/cuda/tests/nocuda/test_library_lookup.py [hunk 1]
```diff
@@ -282,3 +284,24 @@ def _fake_non_conda_env():
...
+        default = "the default"
+
+        # Check that we are getting the default
+        got = envvars.get_numba_envvar(env, default=default)
+        self.assertIs(got, default)
+
+        # Check that we are reading the env-var correctly
+        with override_env_config(f"NUMBA_{env}", "my value"):
+            got = envvars.get_numba_envvar(env, default=default)
+            self.assertEqual(got, "my value")
```
Comment (stuartarchibald) [inline numba/cuda/tests/nocuda/test_library_lookup.py:307]: This seems to be missing:
```
if __name__ == '__main__':
    unittest.main()
```

**Comment by stuartarchibald** [comment]
Patch looks good, one comment about missing `unittest.main()`, would you like to resolve that whilst this file is being changed?

File: numba/cuda/tests/nocuda/test_library_lookup.py (hunk 1 above)
Comment (sklam) [reply numba/cuda/tests/nocuda/test_library_lookup.py:307]: added

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch and fixes.

**Pull request merged**